src/app/ReadHandler.cpp

```diff
@@ -652,6 +719,52 @@ CHIP_ERROR ReadHandler::ProcessSubscribeRequest(System::PacketBufferHandle && aP
 
     mExchangeCtx->WillSendMessage();
 
+#if CHIP_CONFIG_PERSIST_SUBSCRIPTIONS
+    auto * subscriptionResumptionStorage = InteractionModelEngine::GetInstance()->GetSubscriptionResumptionStorage();
+    if (!subscriptionResumptionStorage)
+    {
+        return CHIP_NO_ERROR;
+    }
+
+    SubscriptionResumptionStorage::SubscriptionInfo subscriptionInfo = { .mNodeId         = GetInitiatorNodeId(),
+                                                                         .mFabricIndex    = GetAccessingFabricIndex(),
+                                                                         .mSubscriptionId = mSubscriptionId,
+                                                                         .mMinInterval    = mMinIntervalFloorSeconds,
+                                                                         .mMaxInterval    = mMaxInterval,
+                                                                         .mFabricFiltered = IsFabricFiltered() };
+    ObjectList<AttributePathParams> * attributePath                  = mpAttributePathList;
+    size_t attributePathCount                                        = 0;
+    while (attributePath)
+    {
+        attributePathCount++;
+        attributePath = attributePath->mpNext;
+    }
+    attributePath = mpAttributePathList;
+    subscriptionInfo.mAttributePaths.Calloc(attributePathCount);
+    for (size_t i = 0; i < attributePathCount; i++)
+    {
+        subscriptionInfo.mAttributePaths[i].SetValues(attributePath->mValue);
+        attributePath = attributePath->mpNext;
+    }
+
+    ObjectList<EventPathParams> * eventPath = mpEventPathList;
+    size_t eventPathCount                   = 0;
+    while (eventPath)
+    {
+        eventPathCount++;
+        eventPath = eventPath->mpNext;
+    }
+    eventPath = mpEventPathList;
+    subscriptionInfo.mEventPaths.Calloc(eventPathCount);
```

Comment:
How is a potential allocation failure handled? Or is this guaranteed to succeed? If guaranteed to be ok, please add a comment why.

Reply:
Good point - I've added nullptr checks